src/exec/engine.go

```diff
@@ -179,13 +140,14 @@ func selectProvider(ps []providers.Provider, timeout time.Duration) (providers.P
 		}()
 	}
 
-	var provider providers.Provider
 	select {
-	case provider = <-online:
-		return provider, nil
-	case <-done:
-		return nil, ErrNoProviders
+	case on := <-online:
+		if on {
+			return nil
+		} else {
```

Comment:
Just a style thing. I prefer to see all of the branches next to each other. It doesn't look like we use this elsewhere though. Changing...